source/acquaman/IDEAS/IDEAS2DScanActionController.cpp

```diff
@@ -0,0 +1,199 @@
+		detector->setIsVisible(false);
+		detector->setHiddenFromUsers(true);
+		detectors.addDetectorInfo(detector->toInfo());
+	}
+
+	configuration_->setDetectorConfigurations(detectors);
+
+	secondsElapsed_ = 0;
+	secondsTotal_ = configuration_->totalTime();
+	elapsedTime_.setInterval(1000);
+	connect(this, SIGNAL(started()), &elapsedTime_, SLOT(start()));
+	connect(this, SIGNAL(cancelled()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(paused()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(resumed()), &elapsedTime_, SLOT(start()));
+	connect(this, SIGNAL(failed()), &elapsedTime_, SLOT(stop()));
+	connect(this, SIGNAL(finished()), &elapsedTime_, SLOT(stop()));
+	connect(&elapsedTime_, SIGNAL(timeout()), this, SLOT(onScanTimerUpdate()));
+}
+
+IDEAS2DScanActionController::~IDEAS2DScanActionController()
+{
+
+}
+
+void IDEAS2DScanActionController::buildScanControllerImplementation()
+{
+	AMXRFDetector *detector = 0;
+
+	if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ketek))
+		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("KETEK"));
+
+	else if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ge13Element))
+		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("13-el Ge"));
+
+	if (detector){
+
+		detector->removeAllRegionsOfInterest();
+
+		QList<AMDataSource *> i0Sources = QList<AMDataSource *>() << scan_->dataSourceAt(scan_->indexOfDataSource("I_0"));
+
+		AMDataSource *spectraSource = scan_->dataSourceAt(scan_->indexOfDataSource(detector->name()));
+
+		foreach (AMRegionOfInterest *region, configuration_->regionsOfInterest()){
+
+			AMRegionOfInterestAB *regionAB = (AMRegionOfInterestAB *)region->valueSource();
+			AMRegionOfInterestAB *newRegion = new AMRegionOfInterestAB(regionAB->name().remove(' '));
+			newRegion->setBinningRange(regionAB->binningRange());
+			newRegion->setInputDataSources(QList<AMDataSource *>() << spectraSource);
+			scan_->addAnalyzedDataSource(newRegion, false, true);
+			detector->addRegionOfInterest(region);
+
+			AM2DNormalizationAB *normalizedRegion = new AM2DNormalizationAB(QString("norm_%1").arg(newRegion->name()));
+			normalizedRegion->setInputDataSources(QList<AMDataSource *>() << newRegion << i0Sources);
+			normalizedRegion->setDataName(newRegion->name());
+			normalizedRegion->setNormalizationName(i0Sources.first()->name());
+			scan_->addAnalyzedDataSource(normalizedRegion, true, false);
+		}
+	}
+}
+
+void IDEAS2DScanActionController::createAxisOrderMap()
+{
+	axisOrderMap_.insert(scan_->rawData()->scanAxisAt(0).name, 1);
+	axisOrderMap_.insert(scan_->rawData()->scanAxisAt(1).name, 0);
+}
+
+AMAction3 * IDEAS2DScanActionController::createInitializationActions()
+{
+	AMListAction3 *initializationActions = new AMListAction3(new AMListActionInfo3("IDEAS XAS Initialization Stage 1", "IDEAS XAS Initialization Stage 1"), AMListAction3::Parallel);
```

Comment:
Would it be better to alter these string values to 'IDEAS 2D Scan Initialization...' ?